### PR TITLE
Better API Client errors

### PR DESCRIPTION
**Pull request by angetheo**
- Use #fetch instead of brackets to fall back to default
- Remove -Error from error class names.

### CODE REVIEW

File: spec/routemaster/integration/api_client_spec.rb [hunk 1]
```diff
@@ -51,39 +51,39 @@ class ProcHandler
 
   describe 'error handling' do
     it 'raises an ResourceNotFoundError on 404' do
```
Comment (krissy) [inline spec/routemaster/integration/api_client_spec.rb]: You might want to update the spec descriptions here too, otherwise LGTM!

**Review by krissy — APPROVED** [review]

File: spec/routemaster/integration/api_client_spec.rb (hunk 1 above)
Comment (angetheo) [reply spec/routemaster/integration/api_client_spec.rb]: nice catch, thanks 👍

**Comment by kapowaz** [comment]
img

**Comment by nronas** [comment]
LGTM

**Pull request merged**